Site icon as image & Assignment list

GrahamQuince opened the pull request:
Two snippets

Site icon displays the focussed site's icon as an image

Assignment list returns the current open assignments for a student in a table, without them having to open the assignments app

GrahamQuince committed:
Merge pull request #5 from frogeducation/master

Update

GrahamQuince committed:
Site icon as image & Assignment list

Two snippets:

Site icon displays a site's icon as its image

Assignment List lists out the assignments for a student without them
accessing the app

GrahamQuince committed:
Merge pull request #6 from GrahamQuince/Site-Icon-as-Image

Site icon as image & Assignment list

Smudge3806 commented on site-icon-as-image/snippet.html:
```diff
@@ -0,0 +1,18 @@
+		link: siteLink,
+		method: "sites.getInfo"
+	}).done(function(siteResponse) {
+		$('#siteIcon').html("<img src='"+siteResponse.data.icon+"'>");
```
Use of ID's in FrogOS is prohibited unless the ID is truly unique (contains a timestamp or uses `Frog.Utilities.generateSimpleId()`)

Smudge3806 commented on site-icon-as-image/snippet.html:
```diff
@@ -0,0 +1,18 @@
+<div id="siteIcon"></div>
```
Use of ID's in FrogOS is prohibited unless the ID is truly unique (contains a timestamp or uses `Frog.Utilities.generateSimpleId()`)

Smudge3806 commented on site-icon-as-image/snippet.html:
```diff
@@ -0,0 +1,18 @@
+
+<script>
+  
+var siteLink = (s = $('.ui-dialog.focused').find('.sites_core').data('controllers')) &&
```
`s` is not declared and will therefore leak into the global scope. This is Bad

Smudge3806 replied:
Consider using the HTML widget as a starting point.
`var siteLink = (s = arguments[0].closest('.ui-dialog').find('.sites_core').data('controllers')) &&`

Smudge3806 commented on site-icon-as-image/snippet.html:
```diff
@@ -0,0 +1,18 @@
+<script>
+  
+var siteLink = (s = $('.ui-dialog.focused').find('.sites_core').data('controllers')) &&
+	(c = s.sites_core) &&
```
`c` is not declared and will therefore leak into the global scope. This is Bad

Smudge3806 commented on assignments-list/snippet.html:
```diff
@@ -0,0 +1,128 @@
+<style>
```
This file does not match the one described in the README.

It appears to be a duplicate of the Filedrop slideshow

GrahamQuince committed:
Added correct code for assignments

and improved site icon code based on feedback

GrahamQuince committed:
Merge pull request #7 from GrahamQuince/Site-Icon-as-Image

Added correct code for assignments

Smudge3806 commented on assignments-list/snippet.html:
```diff
@@ -0,0 +1,54 @@
+        limit: 5,
+        offset: 0
+    }).done(function(assignmentResponse) {
+        console.log(assignmentResponse.data.assignments);
```
Should not be using Console.log in production facing code.

Smudge3806 commented on assignments-list/snippet.html:
```diff
@@ -0,0 +1,54 @@
+        console.log(assignmentResponse.data.assignments);
+        assignmentResponse.data.assignments.forEach(function(work, index) {
+
+            var dueDate = new Date(work.assignment.end * 1000).toDateString();
```
Use moment.js. It will be more declarative

Smudge3806 commented on site-icon-as-image/snippet.html:
```diff
@@ -0,0 +1,26 @@
+	}).done(function(siteResponse) {
+		$newDiv.html("<img src='"+siteResponse.data.icon+"'>");
+});
+arguments[0].append($newDiv);
```
Probably worth adding a comment to explain what this is...

Smudge3806 commented on assignments-list/snippet.html:
```diff
@@ -0,0 +1,54 @@
+
+
+Frog.Model
+    .api('assignmentspupils.getmyassigned', {
```
assignmentpupils.getmyassigned is about to be deprecated and removed from the codebase.

Smudge3806 commented:
Formatting is a little chaotic... Couple of Coding conventions that need covering off... Not a million miles from being ready.

Actions required:
- [ ] Neaten code to follow FrogOS Style Guide (Airbnb JS Style Guide)
- [ ] Convert Date manipulation to use moment.js (Documentation in DevDocs
- [ ] Consider changing assignmentpupils.getassigned to newer endpoint.
- [ ] Consider creating HTML content either via use of Template element OR incremental jQuery creation.

GrahamQuince committed:
Updated based on feedback

Site icon is commented, assignments list rewritten for student accessing
latest assignments

GrahamQuince committed:
Merge pull request #8 from GrahamQuince/Site-Icon-as-Image

Updated based on feedback

GrahamQuince committed:
Updated to tidy up styling and add comments

GrahamQuince committed:
Merge pull request #9 from GrahamQuince/Site-Icon-as-Image

Updated to tidy up styling and add comments
